### PR TITLE
Improves support for collada models with non-integer joint indices

### PR DESCRIPTION
**Pull request by ferrolho**
Concerning #12059,

Sorry for the delay committing this, @Mugen87... I went for vacation and completely forgot to push.

I have changed a couple of lines of code so that joint ids of collada models are stored in a `map` rather than in an integer array (some models like the one I added from ABB do use `strings` to identify joints).
In addition, I also changed a bit the tween setup and joint updates for the example demo itself.

I tested with many different robot collada models and everything seems to work fine. The old demo with the Kawada HiroNX still works. I hope I didn't miss/mess anything!

### CODE REVIEW

**Comment by Mugen87** [comment]
For testing: http://rawgit.com/ferrolho/three.js/dev/examples/webgl_loader_collada_kinematics.html

**Comment by Mugen87** [comment]
Looks great!

**Review by Mugen87 — APPROVED** [review]

File: examples/js/loaders/ColladaLoader2.js [hunk 1]
```diff
@@ -2452,7 +2452,9 @@ THREE.ColladaLoader.prototype = {
 					case 'axis':
 						var param = child.getElementsByTagName( 'param' )[ 0 ];
 						data.axis = param.textContent;
-						data.jointIndex = parseInt( data.axis.split( 'joint' ).pop().split( '.' )[ 0 ] );
+						var tmpJointIndex = data.axis.split( 'inst_' ).pop().split( 'axis' )[ 0 ];
+						tmpJointIndex = tmpJointIndex.substr(0, tmpJointIndex.length - 1);
```
Comment (Mugen87) [inline examples/js/loaders/ColladaLoader2.js]: Just a code style issue:
```javascript
tmpJointIndex.substr( 0, tmpJointIndex.length - 1 );
```

Comment (Mugen87) [reply examples/js/loaders/ColladaLoader2.js]: BTW: Can you directly assign this statement to `data.jointIndex`? Then you would save the next line...

**Review by Mugen87 — CHANGES_REQUESTED** [review]

**Comment by Mugen87** [comment]
I think we should change this [line](https://github.com/ferrolho/three.js/blob/bf63085e7181df755d0ad932ebb4892f72b28b81/examples/js/loaders/ColladaLoader2.js#L2168) to `joints: {},`.

File: examples/js/loaders/ColladaLoader2.js (hunk 1 above)
Comment (ferrolho) [reply examples/js/loaders/ColladaLoader2.js]: Sure can! :+1:

**Comment by ferrolho** [comment]
Last commit should contain all the corrections:
- `joints: [],` → `joints: {},`;
- Code style fix;
- Save one line with the direct assignment.

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**